lib/broadway/topology/batcher_stage.ex

```diff
@@ -11,7 +11,7 @@ defmodule Broadway.Topology.BatcherStage do
       __MODULE__,
       args[:processors],
       args,
-      [max_demand: args[:batch_size]],
+      [max_demand: args[:max_demand] || args[:batch_size]],
```

Comment:
Should we raise if no max_demand is given and batch_size is a function?

Reply:
Sorry that I had the assumption that the library user should pass :max_demand when they are passing a function type :batch_size parameter.  Will add the checking.